### ISSUE TITLE
Accessibility: UI component triggers change of context without advanced notice

### ISSUE THREAD

**Issue by vikrampm1** [woothemes/woocommerce#52086]
**Prerequisites (mark completed items with an [x]):**
- [x] I have checked that my issue type is not listed here https://github.com/woocommerce/woocommerce/issues/new/choose
- [x] My issue is not a security issue, support request, bug report, enhancement or feature request (Please use the link above if it is).

### Issue Description:

The default state of the Filter by Attribute block is to not have an Apply button and, instead, the filter submits and the page refreshes when an item is selected in the combo box or checkbox. There is no warning for screen reader users that this will happen. When the page refreshes, keyboard focus is lost and then user is back at the top of the page. There is no auditory message that the products shown on the page have changed.

This is a WCAG failure of 3.2.2 On Input: https://www.w3.org/WAI/WCAG21/Understanding/on-input.html 

![Image](https://github.com/user-attachments/assets/da3c261c-d054-481d-af87-9aff769293ce)

![Image](https://github.com/user-attachments/assets/6c30bbe5-7dd1-46d8-b695-4260058e845a)

![Image](https://github.com/user-attachments/assets/5cb11151-2a97-40bd-a0fa-4a9295cccd4b)

### Component or Page: 

Component: Filter by attribute block

### Name affected element:

Filter by attribute block without the Apply button

### Relevant code snippets:

n/a

### Recommended fix:

A sufficient fix to this failure whould be to add visible or screen reader text that describes the functionality before the filter in the DOM. It could be connected to the filter with an aria-describedby attribute. A simple sentence such as ""Selecting an option will submit the filter and refresh the page"" will work. 

See: https://www.w3.org/WAI/WCAG21/Techniques/general/G13 

A better fix will warn the user and also not cause them to lose keyboard focus on selection of an option. Either submit the filter with ajax or, if keeping the page reload, use JavaScript to shift the focus back to the filter that triggered the change.

There should be an auditory annoucement, "Blue color filter applied, 10 products found"

**Comment by Manussakis** [comment]
I went for the simplest solution:

> A sufficient fix to this failure whould be to add visible or screen reader text that describes the functionality before the filter in the DOM. It could be connected to the filter with an aria-describedby attribute. A simple sentence such as ""Selecting an option will submit the filter and refresh the page"" will work.

---

These are the reasons I didn't implement the other fixes suggested in the issue description:

> A better fix will warn the user and also not cause them to lose keyboard focus on selection of an option. Either submit the filter with ajax...

Moving from a page refreshing to AJAX seems overkill and prone to regressions. There are other filters that also refresh the page once applied:
- woocommerce/active-filters
- woocommerce/price-filter
- woocommerce/stock-filter
- woocommerce/rating-filter

>  ...or, if keeping the page reload, use JavaScript to shift the focus back to the filter that triggered the change.

This means preserving a reference of the filter element across page reloads so it can be focused once the loading is completed.
All of the blocks listed above would need to have access to the filter element that triggered the page reload.
This also seems a huge effort.

> There should be an auditory annoucement, "Blue color filter applied, 10 products found"

Stores can have many filter combinations. Ideally, all of the filters applied should be combined into a single message to be announced for screen reader users. The https://github.com/woocommerce/woocommerce/issues/52090 issue was already created to address it.
In the meantime, users can still tab through the filters checking which of them are applied (I agree it's not the fastest way to get the information).
